src/ioGrowthAverage.cpp

```diff
@@ -0,0 +1,86 @@
+/**
+    ioAverage.cpp
+    Purpose: Program that reads txt file and outputs average growth of methods amplitude and Bits Required and NC
+
+    @author Jorge Miguel Ferreira da Silva
+    @version 0.1
+*/
+
+#include <iostream>     // std::cout
+#include <fstream>      // std::fstream
+#include <string>       // std::string
+#include <sstream>      // std::istringstream
+#include <vector>
+#include <numeric>      //std::accumulate
+#include <algorithm>
+#include <cmath>
+#include <iomanip>
+#include"tmId.h"
+
+
+struct AverageGrowth{
+    unsigned int index;
+    double amp;
+    double nmvc;
+    double nc;
+};
+
+int main(int argc, char *argv[])
+{   
+    
+    const std::string& input= argv[1];
+    std::ifstream in;
+    in.open(input);
+    std::string line;
+    std::vector<AverageGrowth> vector_results;
+    AverageGrowth avg;
+    if(in.is_open())
+    {
+        while(std::getline(in, line)) //get 1 row as a string
+        {
+            //std::cout << line << std::endl;
+            std::istringstream iss(line); //put line into stringstream
+            
+            while(iss >> avg.index >> avg.amp >> avg.nmvc >> avg.nc) //read word by word
+            {   
+                vector_results.push_back(avg);
+            }
+        }
+    }
+
+    std::sort(vector_results.begin(),vector_results.end(), [](const AverageGrowth &x, const AverageGrowth &y){ return (x.index < y.index);});
+    unsigned int value = vector_results[0].index;
+    AverageGrowth growth;
+    AverageGrowth average_growth;
+
+    std::vector <AverageGrowth> results_vector;
+    auto counter = 0;
+    for (auto a = 0u; a<=vector_results.size(); ++a){
```

Comment:
Since you never use the index `a`, you can use the for-each notation and replace all uses accordingly.

```c++
for (auto& result: vector_results) {
    if (result.index == value) {
...
```